crates/cli/src/datadog_utils.rs

```diff
@@ -57,42 +58,68 @@ pub fn get_datadog_variable_value(variable: &str) -> anyhow::Result<String> {
     Err(anyhow!("cannot find variable DD_{}", variable))
 }
 
-// if we put the first argument to true staging, override the value and use staging.
-// otherwise, use the DD_SITE variable or the default site
-fn get_datadog_site(use_staging: bool) -> String {
-    if use_staging {
-        STAGING_DATADOG_SITE.to_string()
+// If a DD_HOSTNAME envvar has been specified, use it; otherwise, use the staging hostname
+// if use_staging is true; otherwise, use the DD_SITE envvar with 'api.' prepended;
+// otherwise, use the default hostname.
+fn get_datadog_hostname(use_staging: bool) -> String {
+    if let Ok(hostname) = get_datadog_variable_value("HOSTNAME") {
+        hostname
+    } else if use_staging {
+        STAGING_DATADOG_HOSTNAME.to_string()
+    } else if let Ok(site) = get_datadog_variable_value("SITE") {
+        format!("api.{}", site).to_string()
```

Comment:
I do not understand this change. Why not just passing `DD_SITE` when invoking the binary? It seems to be doing the same thing? Adding `api` to the site is a breaking change, it would not comply with the docs and conventions we have.

Reply:
Prepending `api.` is the current behavior. `DD_SITE` only provides the part that comes after `api.`. We need `DD_HOSTNAME` because the internal endpoints don't start with `api.`.

Reply:
my comment was rather that there are two concerns at play here: detect the site and detect the hostname. And we put everything in a single function. I would have kept a function to get the site and add another function to get the hostname (that calls the function to get the site). This separates concerns.

Might be overdoing it and not a blocker.